src/rcms/fm/app/level1/HCALlevelTwoEventHandler.java

```diff
@@ -452,135 +457,109 @@ public void configureAction(Object obj) throws UserActionException {
-          if(isSinglePartition){
-            CheckAndSetParameter( parameterSet , "HCAL_ICICONTROL_SINGLE"     ,false);
-            CheckAndSetParameter( parameterSet , "HCAL_PICONTROL_SINGLE"      ,false);
-          }
-          else{
-            CheckAndSetParameter( parameterSet , "HCAL_LPMCONTROL"           ,false);
-            CheckAndSetParameter( parameterSet , "HCAL_ICICONTROL_MULTI"     ,false);
-            CheckAndSetParameter( parameterSet , "HCAL_PICONTROL_MULTI"      ,false);
-          }
         }
         catch (UserActionException e){
           String warnMessage = "[HCAL LVL2 " + functionManager.FMname + "] ConfigureAction: "+e.getMessage();
           logger.warn(warnMessage);
         }
       }
 
-      // Fill the local variable with the value received from LV1
+      // Parse the mastersnippet 
+      String selectedRun       = ((StringT)functionManager.getHCALparameterSet().get("RUN_CONFIG_SELECTED").getValue()).getString();
       CfgCVSBasePath           = ((StringT)functionManager.getHCALparameterSet().get("HCAL_CFGCVSBASEPATH").getValue()).getString();
-      FullCfgScript            = ((StringT)functionManager.getHCALparameterSet().get("HCAL_CFGSCRIPT"     ).getValue()).getString();
+      // Reset HCAL_CFGSCRIPT:
+      functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>("HCAL_CFGSCRIPT",new StringT("not set")));
+      // Try to find a common masterSnippet from MasterSnippet
+      String CommonMasterSnippetFile ="";
+      try{
+        String TagName="CommonMasterSnippet";
+        String attribute="file";
+        CommonMasterSnippetFile = xmlHandler.getHCALMasterSnippetTagAttribute(selectedRun,CfgCVSBasePath,TagName,attribute);
+      }
+      catch(UserActionException e){
+        logger.error("[HCAL LVL2"+functionManager.FMname+"]: Found more than one CommonMasterSnippet tag in the mastersnippet! This is not allowed!");
+        functionManager.goToError(e.getMessage());
+      }
+      if(!CommonMasterSnippetFile.equals("")){    
+          //parse and set HCAL parameters from CommonMasterSnippet
+          logger.info("[HCAL LVL2 "+ functionManager.FMname +"] Going to parse CommonMasterSnippet : "+ CommonMasterSnippetFile);
+          xmlHandler.parseMasterSnippet(CommonMasterSnippetFile,CfgCVSBasePath);
+      }
+      //Parse and set HCAL parameters from MasterSnippet
+      logger.info("[HCAL LVL2 "+ functionManager.FMname +"] Going to parse MasterSnippet : "+ selectedRun);
+      xmlHandler.parseMasterSnippet(selectedRun,CfgCVSBasePath);
+
+      if(!CommonMasterSnippetFile.equals("")){    
+          //parse and set HCAL parameters from CommonMasterSnippet
+          logger.info("[HCAL LVL2 "+ functionManager.FMname +"] Going to parse CommonMasterSnippet(partition specific) : "+ CommonMasterSnippetFile);
+          xmlHandler.parseMasterSnippet(CommonMasterSnippetFile,CfgCVSBasePath,functionManager.FMpartition);
```

Comment:
Parse the commonMS and MS for the 2nd time, looking for only partition-specific elements